src/Blocks/wave_functions.jl

```diff
@@ -0,0 +1,131 @@
+# Define and register smooth functions
+using ModelingToolkit, Symbolics
+_constant(K) = K
+_cos_wave(x, f, A, st, ϕ) = A*cos(2*π*f*(x-st) + ϕ)
+_damped_sine_wave(x, f, A, st, ϕ, d) = exp((st-x)*d)*A*sin(2*π*f*(x-st) + ϕ)
+_ramp(x, δ, st, et, h) = h/(et-st)*(_xH(x, δ, st) - _xH(x, δ, et))
+_sine(x, f, A, st, ϕ) = A*sin(2*π*f*(x - st) + ϕ)
+_square_wave(x, δ, f, A, st) = A*2atan(sin(2π*(x-st)*f)/δ)/π
+_step(x, δ, h, a) = h*(atan((x-a)/δ)/π + 0.5)
```

Comment:
```suggestion
_step(x, δ, h, a) = h*(atan((x-a)/δ)/π + 1//2)
```
something like this would perhaps be nice to not auto promote to `Float64`